gax/src/main/java/com/google/api/gax/retrying/ExponentialRetryAlgorithm.java

```diff
@@ -100,19 +100,31 @@ public TimedAttemptSettings createNextAttempt(TimedAttemptSettings prevSettings)
           (long) (settings.getRetryDelayMultiplier() * prevSettings.getRetryDelay().toMillis());
       newRetryDelay = Math.min(newRetryDelay, settings.getMaxRetryDelay().toMillis());
     }
+    Duration randomDelay = Duration.ofMillis(nextRandomLong(newRetryDelay));
 
     // The rpc timeout is determined as follows:
     //     attempt #0  - use the initialRpcTimeout;
-    //     attempt #1+ - use the calculated value.
+    //     attempt #1+ - use the calculated value or the time remaining in totalTimeout.
```

Comment:
Please provide more details under which condition each variant is chosen (otherwise it sounds like a non-deterministic choice between the two)

Reply:
Good call. Done.